source/agora/common/DNS.d

```diff
@@ -399,6 +399,16 @@ public struct Header
     /// Reserved for future use. Must be zero in all queries and responses.
     public ubyte Z () const scope { return (this.field2 & 0b0_0000_0_0_0_0_111_0000) >> 4; }
 
+    /// Authenticated Data, initially defined in RFC2065
+    public bool AD () const scope { return !!(this.field2 & (1 << 5)); }
+    /// Ditto
+    public void AD (bool val) scope { this.field2 = (this.field2 & ~(1 << 5)) | (1 << 5); }
+
+    /// Checking Disabled, initially defined in RFC2065
+    public bool CD () const scope { return !!(this.field2 & (1 << 4)); }
+    /// Ditto
+    public void CD (bool val) scope { this.field2 = (this.field2 & ~(1 << 4)) | (1 << 4); }
```

Comment:
So both getter and setter knows `CD` is on 4th bit, should be store this info on a mapping (like static enum)?

Reply:
I think it would make sense to have a `mixin` for this.